kernel/ixgbe/src/lib.rs

```diff
@@ -0,0 +1,1235 @@
+//! An ixgbe driver for a 82599 10GbE Network Interface Card.
+//! 
+//! Currently we support basic send and receive, Receive Side Scaling (RSS), 5-tuple filters, and MSI interrupts. 
+//! We also support language-level virtualization of the NIC so that applications can directly access their assigned transmit and receive queues.
+//! When using virtualization, we disable RSS since we use 5-tuple filters to ensure packets are routed to the correct queues.
+//! We also disable interrupts, since we do not yet have support for allowing applications to register their own interrupt handlers.
+//! 
+//! Before using the driver, make sure to set the link type as either 1GB or 10GB by setting the `IXGBE_10GB_LINK` variable.
```

Comment:
I can improve the init function to take certain parameters for each NIC (like which features are enabled), but resetting them on the fly is a bit more complicated. I would have to see the procedure from the reference manual because it's possible that some changes would require a software reset.
For now I'll just remove some of these constants and make them options at the time of initializing the NIC.

Reply:
That's totally fine. For now, accept parameters for only the config options you know you can change dynamically, for others, specify them as consts but split them off into a separate section of the code that makes it clear they are to be modified by a developer.